_util/deep_assign.ts

```diff
@@ -30,6 +30,10 @@ object | undefined {
         target[key] = new Date(value);
         return;
       }
+      if (value instanceof RegExp) {
+        target[key] = new RegExp(value);
+        return;
+      }
```

Comment:
I'm not quite happy with the fix as it seems `deep_assign` is still flawed. I'm sure sure about its usecases, given it lives in a `_util` directory… Looks like it is supposed to be something private?

Anyway, as fixing this implementation in particular is not the goal of this PR, I guess we could come back to it later.